### PR TITLE
[dataquery] Adding the test plan

### PR DESCRIPTION
**Pull request by cmadjar**
## Brief summary of changes

Adds the test plan for the DQT...

#### Testing instructions (if applicable)

1. Follow the test plan

### CODE REVIEW

File: modules/dataquery/test/TestPlan.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+   - ensure that clicking on the different page numbers on the right side are working
+   - change the data view from "Cross-sectional" to "Longitudinal", ensure that the 
+     data is now displayed longitudinal on the X axis with column header names changed
+     to "VisitLabel,InstrumentName" separated with a comma
+6. Go to the "Statistical Analysis" tab:
+   - ensure that a table with "Basic Statistics" (Min, Max, Standard Deviation, Mean,
+     Mean Deviation, Mean Squared Error, First Quartile, Second Quartile, Third 
+     Quartile) is shown and filled
+   - ensure that the "Scatterplot" is working when selecting data for column X and Y
+7. Go to the "Manage Saved Queries" tab:
```
Comment (cmadjar) [inline modules/dataquery/test/TestPlan.md]: @driusan This is not working so I could not really elaborate on this one. Maybe you can help? Or someone else with the DQT setup?

File: modules/dataquery/test/TestPlan.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   - ensure the current query can be saved
+   - ...
+8. Go to the "Load Saved Query" tab:
```
Comment (cmadjar) [inline modules/dataquery/test/TestPlan.md]: @driusan This one either is not working so I could not really elaborate. Maybe you can help? Or someone else with the DQT setup?

File: modules/dataquery/test/TestPlan.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   - ensure that the search of fields within the instrument is working
+   - ensure that clicking on the "Add All" button adds all the fields for the 
+     selected instrument to the list of fields to be queried on the
+     right side of the screen
+   - ensure that clicking on the "Remove All" button removes all the selected 
+     fields for the selected instrument to the list of fields to be queried on the
+     right side of the screen
+   - select a few fields from different instruments and ensure that they are all added 
+     to the list of selected fields on the right side of the screen
+   - ensure that the number of visits selected reflect the number of visits that were
```
Comment (PapillonMcGill) [inline modules/dataquery/test/TestPlan.md]: Maybe rephrase that a bit differently?
```suggestion
   - ensure that the number of visits added to the query reflect the number of visits that were
```

File: modules/dataquery/test/TestPlan.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+     comparison operators appears next to the selected field; select a comparison; 
+     when applicable, ensure that a text box appears next to the comparison operator;
+     enter a value in the text box and ensure a drop down with the list of visits 
+     on which to apply the filter appears next to the text box
+   - add another filter by clicking on the "Add Rule" button
+   - add a group filter by clicking on the "Add Group" button
+   - ensure that the delete button for a given filter deletes the filter
+   - ensure that the delete button for a given group of filters deletes the group of filters
+5. Go to the "View Data" tab:
+   - ensure that a "We are currently working hard to load your data. Please be patient."
```
Comment (PapillonMcGill) [inline modules/dataquery/test/TestPlan.md]:
```
    "We are currently working hard to load your data. Please be patient."
```
Never knew  there were some gnomes inside DQT :-)

File: modules/dataquery/test/TestPlan.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     message is displayed after clicking on the "Run Query" button and before the 
+     data is displayed
+   - ensure that a table with results is being displayed after having run the query
+   - play with the "(Maximum rows per page)" drop down and ensure the table gets updated
```
Comment (PapillonMcGill) [inline modules/dataquery/test/TestPlan.md]:
```suggestion
   - Change the "(Maximum rows per page)" drop down and ensure the table gets updated
```

File: modules/dataquery/test/TestPlan.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+# Data Query Tool (DQT) Test Plan
+
+1. Access the DQT page with a user that has permission 'View Data Query Tool', ensure it renders
```
Comment (johnsaigle) [inline modules/dataquery/test/TestPlan.md]:
```suggestion
1. Ensure the DQT module loads only when with a user that has the permission 'View Data Query Tool' (`dataquery_view`)
```

File: modules/dataquery/test/TestPlan.md (hunk 5 above)
Comment (cmadjar) [reply modules/dataquery/test/TestPlan.md]: haha. Way better phrasing indeed ;)

**Comment by cmadjar** [comment]
@PapillonMcGill @johnsaigle Thank you for the reviews. I updated the test plan accordingly. Please re-review when you get a chance

File: modules/dataquery/test/TestPlan.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
+# Data Query Tool (DQT) Test Plan
+
+1. Ensure the DQT module loads only when with user that has the permission 'View Data Query Tool' (`dataquery_view`)
```
Comment (PapillonMcGill) [inline modules/dataquery/test/TestPlan.md]:
```suggestion
1. Ensure the DQT module loads only for user that has the permission 'View Data Query Tool' (`dataquery_view`)
```
small syntax problem.

Comment (cmadjar) [reply modules/dataquery/test/TestPlan.md]: ooops. fixed in the last commit

File: modules/dataquery/test/TestPlan.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+# Data Query Tool (DQT) Test Plan
+
+1. Ensure the DQT module loads only for user that has the permission 'View Data Query Tool' (`dataquery_view`)
```
Comment (johnsaigle) [inline modules/dataquery/test/TestPlan.md]:
```suggestion
1. Ensure the DQT module loads only for a user that has the permission 'View Data Query Tool' (`dataquery_view`)
```

Comment (cmadjar) [reply modules/dataquery/test/TestPlan.md]: sorry. This time should be good. ;)

**Comment by cmadjar** [comment]
@PapillonMcGill I finalized the test plan in the last commit, could you please re-review? Thank you!

File: modules/dataquery/test/TestPlan.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+6. Go to the "Statistical Analysis" tab:
+   - ensure that a table with "Basic Statistics" (Min, Max, Standard Deviation, Mean,
+     Mean Deviation, Mean Squared Error, First Quartile, Second Quartile, Third 
+     Quartile) is shown and filled
+   - ensure that the "Scatterplot" is working when selecting data for column X and Y
+7. Go to the "Manage Saved Queries" tab:
+   - ensure the 'Save Current Query' button opens to a modal window with a text box 
+     to write the name of the query to be saved, a checkbox to publicly share the 
+     query and 2 buttons at the bottom ('Close' and 'Save Changes') 
+     entering a name for that query and clicking on 'Save Changes' in the modal win
```
Comment (PapillonMcGill) [inline modules/dataquery/test/TestPlan.md]:
```suggestion
     entering a name for that query and clicking on 'Save Changes' in the modal window
```
??

**Comment by ridz1208** [comment]
I wouldnt mind @LeighMac taking a look at this
